File: nodejs/otc-deletebucket.js
var AWS = require('aws-sdk');

var accessKeyId = process.env.S3_ACCESS_KEY_ID; 
var secretAccessKey = process.env.S3_SECRET_ACCESS_KEY; 
var endpoint = process.env.S3_HOSTNAME; 

AWS.config = new AWS.Config({
  accessKeyId: accessKeyId, secretAccessKey: secretAccessKey, region: 'eu-de'
});

var ep = new AWS.Endpoint(endpoint);
var s3 = new AWS.S3({endpoint: ep});  

var myBucket = 'uploadtest';

s3.listObjects({Bucket: myBucket}, function (err, data) {
	if (err) {
		console.log("error listing bucket objects "+err);
		return;
	}
	var items = data.Contents;
	for (var i = 0; i < items.length; i += 1) {
		var deleteParams = {Bucket: myBucket, Key: items[i].Key};
		s3.deleteObject(deleteParams, function (err, data) {
			if (err) {
				console.log("delete err " + deleteParams.Key);
			} else {
				console.log("deleted " + deleteParams.Key);
			}			
		});
	}
});


s3.deleteBucket({Bucket: myBucket}, function (err, data) {	
	if (err) {
		console.log("error deleting bucket " + err);
	} else {
		console.log("delete the bucket " + data);
	}
});

File: python/otc-deletebucket.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# This file is part of OTC S3 Example  released under MIT license.
# Copyright (C) 2016 T-systems Zsolt Nagy

import boto3
from botocore.exceptions import ClientError
import os

ak = os.getenv("S3_ACCESS_KEY_ID", None)
sk = os.getenv("S3_SECRET_ACCESS_KEY", None)       
host = os.getenv("S3_HOSTNAME", None)
region = "eu-de"
		
otcsession = boto3.session.Session()
s3client = otcsession.client('s3', region, 
                           config=boto3.session.Config(signature_version='s3v4'),
                           endpoint_url="https://" + host,
                           aws_access_key_id=ak, 
                           aws_secret_access_key=sk,
                           verify=False
                           )    
myBucket = 'createbuckettest'

try:                                   
	s3client.delete_bucket(Bucket=myBucket) 
except ClientError as e:
    print (str(e))	
except ClientError as e:
	print (str(e))
        
